### PR TITLE
[RUM-8700] Fix Baggage header support 

### PR DESCRIPTION
**Pull request by Valpertui**
### What and why?

This PR introduces W3C baggage header support to the Datadog iOS SDK for `account.id` and `user.id` attributes, enabling seamless propagation of user and account information across distributed systems.  
The feature integrates with the existing tracing infrastructure and provides comprehensive test coverage for edge cases around baggage field key and values validation. Example usage is demonstrated in the sample app showing how to leverage the new account propagation capabilities.

### How?

Implementation Details:
- Core Infrastructure: Extended `TraceContext` with `userId` and `accountId` fields, and updated `W3CHTTPHeadersWriter` to automatically inject baggage headers containing user/account information into outgoing network requests.
- Baggage Header Handling: Implemented `BaggageHeaderMerger` following W3C baggage specification with proper key-value parsing, validation, and merging logic that combines existing baggage headers with new SDK-generated ones.
- Context Propagation: Added `NetworkContext` with `UserConfigurationContext` and `AccountConfigurationContext` to enable seamless propagation of account and user data through the network instrumentation layer.
- Integration Points: Updated `URLSessionRUMResourcesHandler` and `TracingURLSessionHandler` to leverage the new baggage header merging functionality, ensuring compatibility with existing RUM and tracing workflows.

### Review checklist
- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes
- [ ] Add Objective-C interface for public APIs (see our [guidelines](https://datadoghq.atlassian.net/wiki/spaces/RUMP/pages/3157787243/RFC+-+Modular+Objective-C+Interface#Recommended-solution) (internal) and run `make api-surface`)

### CODE REVIEW

**Review by domalessi — APPROVED** [review]

File: DatadogInternal/Sources/NetworkInstrumentation/Datadog/HTTPHeadersWriter.swift [hunk 1]
```diff
@@ -57,8 +57,18 @@ public class HTTPHeadersWriter: TracePropagationHeadersWriter {
...
-                traceHeaderFields[W3CHTTPHeaders.baggage] = "\(W3CHTTPHeaders.Constants.rumSessionBaggageKey)=\(sessionId)"
+                baggageItems.append("\(W3CHTTPHeaders.Constants.rumSessionBaggageKey)=\(sessionId)")
+            }
+            if let userId = traceContext.userId {
+                baggageItems.append("\(W3CHTTPHeaders.Constants.userBaggageKey)=\(userId)")
+            }
+            if let accountId = traceContext.accountId {
+                baggageItems.append("\(W3CHTTPHeaders.Constants.accountBaggageKey)=\(accountId)")
+            }
+            if baggageItems.isEmpty == false {
```
Comment (mariedm) [inline DatadogInternal/Sources/NetworkInstrumentation/Datadog/HTTPHeadersWriter.swift:70]: nit
```suggestion
            if !baggageItems.isEmpty {
```

File: DatadogInternal/Sources/NetworkInstrumentation/NetworkContext.swift [hunk 2]
```diff
@@ -11,4 +11,82 @@ import Foundation
...
+    }
+}
+
+/// The Account configuration context received from `Core`.
+public struct AccountConfigurationContext: AdditionalContext, Equatable {
+    /// Account configuration key in core additional context.
+    public static let key = "account_configuration"
+
+    /// Account ID
+    public let id: String
```
Comment (mariedm) [inline DatadogInternal/Sources/NetworkInstrumentation/NetworkContext.swift:68]: This one is not optional?

**Review by mariedm — APPROVED** [review]
LGTM ✅

File: DatadogInternal/Sources/NetworkInstrumentation/Datadog/HTTPHeadersWriter.swift (hunk 1 above)
Comment (simaoseica-dd) [reply DatadogInternal/Sources/NetworkInstrumentation/Datadog/HTTPHeadersWriter.swift:70]: IMO `== false` helps readability. The `!` operator can be missed when reading quickly.

Comment (Valpertui) [reply DatadogInternal/Sources/NetworkInstrumentation/Datadog/HTTPHeadersWriter.swift:70]: Agree @simaoseica-dd, I miss the leading `!` all the time when reading code

File: DatadogInternal/Sources/NetworkInstrumentation/NetworkContext.swift (hunk 2 above)
Comment (Valpertui) [reply DatadogInternal/Sources/NetworkInstrumentation/NetworkContext.swift:68]: No, an account must have an `id`.  
This is also where we want to bring `user` to, but until the api change is complete we must allow `nil` user.id
